### ISSUE TITLE
Unit testing HttpContext with HTTP2 features

### ISSUE THREAD

**Issue by JamesNK** [aspnet/AspNetCore#6871]
Is there a simple way to create a HttpContext with HTTP2 features?

```cs
var context = new DefaultHttpContext();
context.Response.AppendTrailers("", "");
```

This throws an error because there is no `IHttpResponseTrailersFeature`.

Is there a simple way to make the context HTTP2-erized. If not, consider adding one.

**Comment by Tratcher** [comment]
Yeah, making and setting your own IHttpResponseTrailersFeature ought to be trivial. There's also a IHeadersDictionary implementation in Http you can re-use.

This isn't a feature that DefaultHttpContext should provide by default, you need to know it's going to work before calling it. We could make a dummy implementation of the feature though so you just have to new it up and set it.

**Comment by anurse** [comment]
Duplicate of #6880 ?

**Comment by JamesNK** [comment]
This issue is about using DefaultHttpContext with trailers independently.

#6880 is about using trailers inside TestHost.

**Comment by anurse** [comment]
Since trailers aren't as reliable a feature, I don't think we'd put this in `DefaultHttpContext` directly, we should look at an `.EnableHttp2Goop()` function of some kind to initialize default values for HTTP/2 expected features in `DefaultHttpContext`

**Comment by anurse** [comment]
We'll leave this in preview 7, but it's on the verge :). If you're passionate @JamesNK , we take PRs 🐴
